checker/jdk/nullness/src/java/lang/Class.java

```diff
@@ -54,7 +59,7 @@ protected Class() {}
   public boolean desiredAssertionStatus() { throw new RuntimeException("skeleton method"); }
   @Pure public boolean isEnum() { throw new RuntimeException("skeleton method"); }
   public @NonNull T @Nullable [] getEnumConstants() { throw new RuntimeException("skeleton method"); }
-  java.util.Map<String, T> enumConstantDirectory() { throw new RuntimeException("skeleton method"); }
+  java.util.Map<String, @NonNull T> enumConstantDirectory() { throw new RuntimeException("skeleton method"); }
```

Comment:
Can you add a test case for enumConstantDirectory and why the value is non-null?
Does this work:

<T> Map<String, T> getECD(Class<T> p) {
  return p.enumConstantDirectory();
}